Current/Invalid Objects.sql

```diff
@@ -0,0 +1,59 @@
+USE [SQLCopTests]
+GO
+SET ANSI_NULLS ON
+GO
+SET QUOTED_IDENTIFIER ON
+GO
+--  Test to identify Invalid Stored Procedures and Views
+--  If any invalid objects are found then fail the test with list of affected objects
+--	If you require a comprehensive list of Invalid Objects you can use the SQL Prompt find invalid objects functionality: https://documentation.red-gate.com/sp/sql-refactoring/refactoring-an-object-or-batch/finding-invalid-objects
+
+--	Written by Chris Unwin 17/09/2019
+
+ALTER PROCEDURE [SQLCop].[test Invalid Objects]
+AS
+BEGIN
+
+    SET NOCOUNT ON;
+
+    --Assemble
+    -- Declare and set output
+
+    DECLARE @Output VARCHAR(MAX);
+    SET @Output = '';
+
+    -- Act
+    -- Fetch all invalid objects from sys.sql_expression_dpenedencies and write to output
+
+    SELECT @Output
+        = @Output + 'Invalid ' + (CASE
+                                      WHEN ob.type = 'P' THEN
+                                          'stored procedure '
+                                      ELSE
+                                          CASE
+                                              WHEN ob.type = 'V' THEN
+                                                  'view '
+                                              ELSE
+                                                  'object type ' + ob.type + ' '
+                                          END
+                                  END
+                                 ) + '[' + SCHEMA_NAME(ob.schema_id) + '].[' + OBJECT_NAME(dep.referencing_id)
+          + '] relies on missing object [' + dep.referenced_schema_name + '].[' + dep.referenced_entity_name + ']'
+          + CHAR(13) + CHAR(10)
+    FROM sys.sql_expression_dependencies dep
```

Comment:
It looks like `sys.sql_expression_dependencies` was introduced in SQL Server 2016, so this won't work in earlier version. This may or may not be a problem, depending on what the minimum supported SQL Server version is for SQLCop.

https://docs.microsoft.com/en-us/sql/relational-databases/system-catalog-views/sys-sql-expression-dependencies-transact-sql?view=sql-server-2017&viewFallbackFrom=sql-server-2014

Reply:
Actually, I've found references online to `sys.sql_expression_dependencies` that pre-date SQL Server 2016, so I'm not sure how far back this will work. Might be worth digging into further.

Reply:
Never mind. I've found it referenced in SQL Server 2008. This is a non-issue. Soz!